# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1311107089 25200
#      Tue Jul 19 13:24:49 2011 -0700
# Node ID 8f9702abe3e3419c36d1fc6cce6df4d497ec6daf
# Parent  a9d292ce548963afdca0071a361f27b4a82be600
test: Add validation for allow_noninteger_range_as_index()

* ov-range.cc: Add validation for allow_noninteger_range_as_index()

diff --git a/src/ov-range.cc b/src/ov-range.cc
--- a/src/ov-range.cc
+++ b/src/ov-range.cc
@@ -650,8 +650,21 @@ DEFUN (allow_noninteger_range_as_index, 
 Query or set the internal variable that controls whether non-integer\n\
 ranges are allowed as indices.  This might be useful for @sc{matlab}\n\
 compatibility; however, it is still not entirely compatible because\n\
 @sc{matlab} treats the range expression differently in different contexts.\n\
 @end deftypefn")
 {
   return SET_INTERNAL_VARIABLE (allow_noninteger_range_as_index);
 }
+
+/*
+%!test
+%! x = 0:10;
+%! save = allow_noninteger_range_as_index (0);
+%! fail ('x(2.1:5)');
+%! assert (x(2:5), 1:4);
+%! allow_noninteger_range_as_index (1);
+%! assert (x(2.49:5), 1:3);
+%! assert (x(2.5:5), 2:4);
+%! assert (x(2.51:5), 2:4);
+%! allow_noninteger_range_as_index (save);
+*/
